### PR TITLE
Move GenAI metrics / examples into GenAI namespace

### PR DESCRIPTION
**Pull request by dbczumar**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/dbczumar/mlflow/pull/10138?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10138/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10138
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Put EvaluationExample in genai namespace

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [x] Yes. I've updated:
  - [ ] Examples
  - [x] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/metrics/genai/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+from mlflow.metrics.genai.base import EvaluationExample
+from mlflow.metrics.genai.metric_definitions import *
```
Comment (dbczumar) [inline mlflow/metrics/genai/__init__.py:2]: Needed for https://github.com/mlflow/mlflow/pull/10138/files#diff-8f6c5bf82b3cd8071b378ed3d54f25ae369c17319ce4e1d4fd95861de9fe3b2aR588

File: docs/source/python_api/mlflow.metrics.rst [hunk 2]
```diff
@@ -98,26 +98,27 @@ Users create their own :py:class:`EvaluationMetric <mlflow.metrics.EvaluationMet
...
-.. autofunction:: mlflow.metrics.answer_correctness
+.. automodule:: mlflow.metrics
+    :members:
+    :undoc-members:
+    :show-inheritance:
+    :exclude-members: MetricValue, EvaluationMetric, make_metric, EvaluationExample, ari_grade_level, flesch_kincaid_grade_level, rouge1, rouge2, rougeL, rougeLsum, toxicity, answer_similarity, answer_correctness, faithfulness, answer_relevance, mae, mape, max_error, mse, rmse, r2_score, precision_score, recall_score, f1_score, token_count, latency
 
-.. autofunction:: mlflow.metrics.faithfulness
+Generative AI Metrics
+---------------------
```
Comment (dbczumar) [inline docs/source/python_api/mlflow.metrics.rst:108]: @prithvikannan @annzhang-db @sunishsheth2009 let me know what you  think about this naming. Happy to adjust for consistency with the rest of our docs updates.

**Comment by dbczumar** [comment]
@mlflow-automation autoformat

File: mlflow/metrics/__init__.py [hunk 3]
```diff
@@ -435,7 +424,6 @@ def f1_score() -> EvaluationMetric:
     "rougeL",
     "rougeLsum",
     "toxicity",
-    "make_genai_metric",
```
Comment (annzhang-db) [inline mlflow/metrics/__init__.py:438]: Do we also need to remove "answer_similarity", "answer_correctness", etc. here?

File: docs/source/python_api/mlflow.metrics.rst (hunk 2 above)
Comment (annzhang-db) [reply docs/source/python_api/mlflow.metrics.rst:108]: I think this makes sense considering users will be accessing these metrics through `mlflow.metrics.genai.metric_name` and it is consistent with that experience - @sunishsheth2009 @prithvikannan thoughts?

**Review by annzhang-db — APPROVED** [review]
LGTM pending one small change to remove genai metrics from __all__ in mlflow/metrics/__init__.py. 😃

File: docs/source/python_api/mlflow.metrics.rst (hunk 2 above)
Comment (prithvikannan) [reply docs/source/python_api/mlflow.metrics.rst:108]: lgtm!

File: mlflow/metrics/__init__.py (hunk 3 above)
Comment (dbczumar) [reply mlflow/metrics/__init__.py:438]: oof, great catch - fixed!

**Review by sunishsheth2009 — APPROVED** [review]

**Pull request merged**